### PR TITLE
metamask wc deeplink

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

Using wallet connect recommending MetaMask wasn't working.

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

**Review by andrepimenta — APPROVED** [review]

**Review by ibrahimtaveras00 — APPROVED** [review]

**Comment by vonoro** [comment]
@estebanmino could you please advise me on how a url object can both

(1) have `metamask` in `protocol` field and at the same time
(2) have `metamask://wc` in `origin` field (added in this PR)?

https://github.com/MetaMask/metamask-mobile/pull/2038/files#diff-11a6eca2f0edf9a5f087e88e1ee68456686e8ef3c16466e41a5ddf3ee23e0a43R177

I might be stupid but it seems like in this case, the `origin` will always be null and so the condition is never true? See the test cases for `origin` in `url-parse` package https://github.com/unshiftio/url-parse/blob/f7774f608418b59a43ad1816c954654fdc8b1248/test/test.js#L433.

More generally, the PR is kind of vague, it'd be really nice if the PR included some details on how it can be tested 😢 

Basically, what I'm trying to do is open metamask from an android dapp, and wire it up to my dapp using walletConnect, the same way that OpenSea does it. What is the right way to do this on android? 
thanks so much :)

**Comment by vonoro** [comment]
related: https://github.com/MetaMask/metamask-mobile/pull/3120

**Comment by vonoro** [comment]
in case anyone has the same problem, here's what I ended up using

```
        Intent launchIntent = getReactApplicationContext().getPackageManager().getLaunchIntentForPackage(packageId);
        if (launchIntent != null) {
          launchIntent.setData(Uri.parse(wcUri));
          getReactApplicationContext().startActivity(launchIntent);
        }
```
